test/e2e/tests/multi_zone_e2e_test.go

```diff
@@ -141,17 +149,114 @@ var _ = Describe("GCE PD CSI Driver Multi-Zone", func() {
 			i = i + 1
 		}
 	})
-})
 
-type verifyArgs struct {
-	publishDir string
-}
+	It("Should create a RePD instance, write to it, force-attach it to another instance, and read the same data", func() {
+		Expect(testContexts).NotTo(BeEmpty())
+
+		zoneToContext := map[string]*remote.TestContext{}
+		zones := []string{}
+
+		for _, tc := range testContexts {
+			_, z, _ := tc.Instance.GetIdentity()
+			// Zone hasn't been seen before
+			if _, ok := zoneToContext[z]; !ok {
+				zoneToContext[z] = tc
+				zones = append(zones, z)
+			}
+			if len(zoneToContext) == 2 {
+				break
+			}
+		}
 
-type verifyFunc func(verifyArgs) error
+		Expect(len(zoneToContext)).To(Equal(2), "Must have instances in 2 zones")
+
+		controllerContext := zoneToContext[zones[0]]
+		controllerClient := controllerContext.Client
+		controllerInstance := controllerContext.Instance
+
+		p, _, _ := controllerInstance.GetIdentity()
+
+		region, err := common.GetRegionFromZones(zones)
+		Expect(err).To(BeNil(), "Failed to get region from zones")
+
+		// Create Disk
+		volName := testNamePrefix + string(uuid.NewUUID())
+		volID, err := controllerClient.CreateVolume(volName, map[string]string{
```

Comment:
Would it be worth it to also validate that the volumeContext is set correctly, when availability-class is set?

Reply:
good point, thx.